File: v100/ok/10041/program.java
import java.util.*;
import java.io.*;
import java.text.*;

class Main {

	public static void main(String args[]) throws IOException {
	
		int casos = Integer.parseInt(readLine(100));
		while(casos--!=0) {
		
			String l = readLine(2000);
			if(l.trim().equals("")) continue;
			String[] line = split(l);
			double vizinhos = Integer.parseInt(line[0]);
			//double vizinhosOficiais = 0;
			long soma = 0;

			Vector relativos = new Vector();
			for(int i=0;i!=vizinhos;i++) {
				int temp = Integer.parseInt(line[i+1]);
				//if(!relativos.contains(new Integer(temp))) {
					relativos.addElement(new Integer(temp));
				//}
				soma += temp;
			}

			// divide soma por vizinhos
			double media = (soma *1.0) / vizinhos;
			long mediana = 0;
			double menorDist = 9999999;
			for(int i=0;i!=vizinhos;i++) {
				if(Math.abs(((Integer)relativos.elementAt(i)).intValue() - media)<menorDist) {
					menorDist = Math.abs(((Integer)relativos.elementAt(i)).intValue() - media);
					mediana = ((Integer)relativos.elementAt(i)).intValue();
				}
			}

			soma = 0;
			for(int i=0;i!=vizinhos;i++) {
				soma += Math.abs(((Integer)relativos.elementAt(i)).intValue() - mediana);
			}

			System.out.println(soma);

		}

	}
	
	static String[] split(String s) {
		StringTokenizer st = new StringTokenizer(s, " ");
		String[] tokens = new String[st.countTokens()];
		for (int i = 0; i != tokens.length; i++) {
			tokens[i] = st.nextToken();
		}
		return tokens;
	}

	// le uma linha inteira
	static String readLine(int maxLg) throws IOException {

		byte lin[] = new byte[maxLg];
		int lg = 0, car = -1;

		while (lg < maxLg) {
			car = System.in.read();
			if ((car < 0) || (car == '\n')) {
				break;
			}
			lin[lg++] += car;
		}

		if ((car < 0) && (lg == 0)) {
			return (null); // eof
		}

		return (new String(lin, 0, lg));

	}

}

File: v100/ok/10041/program.cpp
#include <stdio.h>
#include <iostream>
#include <vector>
#include <algorithm>
#include <math.h>

using namespace std;

int main(char **argv,int argc) {

	int casos;
	cin >> casos;
	while(casos--) {
		int vizinhos;
		long soma = 0;
		cin >> vizinhos;
		vector<int> relativos;
		for(int i=0;i!=vizinhos;i++) {
			int temp;
			cin >> temp;
		//	if(find(relativos.begin(),relativos.end(),temp)==relativos.end()) {
				relativos.push_back(temp);
		//	}
			soma += temp;
		}
		
		// divide soma por vizinhos
		long media;
		if(soma % vizinhos > (vizinhos/2.0)) {
			media = (soma / vizinhos) + 1;
		} else {
			media = (soma / vizinhos);
		}
		
		//cout << media << endl;
		
		soma = 0;
		for(int i=0;i!=relativos.size();i++) {
			soma += abs(relativos[i] - media);
		}
		
		cout << soma << endl;
		
	}

	return 0;

}
